autogpt_platform/backend/backend/api/features/onboarding_dump/transcription.py

```
@@ -0,0 +1,259 @@
+    """
+    ffmpeg = shutil.which("ffmpeg")
+    if ffmpeg is None:
+        raise TranscriptionFailedError(
+            "ffmpeg is required to transcribe recordings over "
+            f"{SINGLE_REQUEST_MAX_SECONDS // 60} minutes but is not installed"
+        )
+
+    suffix = os.path.splitext(filename)[1] or ".webm"
+    with tempfile.TemporaryDirectory() as workdir:
+        source = os.path.join(workdir, f"source{suffix}")
+        with open(source, "wb") as handle:
```

Ruff (0.16.0)

[warning] Async functions should not open files with blocking methods like `open`

(ASYNC230)